bleach/html5lib_shim.py

```diff
@@ -246,12 +259,14 @@ def __iter__(self):
                      token.get('data'))):
                     # Remove attribute names that have ', " or < in them
                     # because those characters are invalid for attribute names.
-                    token['data'] = [
-                        item for item in token['data']
-                        if ('"' not in item[0] and
-                            "'" not in item[0] and
-                            '<' not in item[0])
-                    ]
+                    token["data"] = {
+                        attr_name: attr_val
+                        for attr_name, attr_val in token["data"].items()
+                        if not any(
+                            invalid_char in attr_name[0]
+                            for invalid_char in INVALID_HTML_ATTR_CHARS
+                        )
+                    }
```

Comment:
I'm not sure I understand this.

`token["data"]` used to be a list of (name, value) tuples, but now it looks like it's a dict of name -> value. 

If that's true, what's `attr_name` look like? Is this looking at the first character of `attr_name` or is `attr_name` a list/tuple?

Reply:
Gotcha, I wasn't clear on why we were pulling the first element from `token["data"]`. Updated to test the full attribute name for invalid chars and preserve the underlying type.

Reply:
It was looking at the first char of `attr_name`, which was wrong.